app/routes.py

```diff
@@ -1,2 +1,314 @@
-from flask import Blueprint
+import re
+from app.models.task import Task
+from app.models.goal import Goal
+from app import db 
+from flask import request, Blueprint, make_response, jsonify#
+from datetime import datetime
+import os
+import requests
+from dotenv import load_dotenv
 
+tasks_bp = Blueprint("tasks", __name__, url_prefix="/tasks") 
+goals_bp = Blueprint("goals", __name__, url_prefix="/goals") 
+
+load_dotenv()
+
+def post_message_to_slack(text):
+    SLACK_TOKEN = os.environ.get('SLACKBOT_TOKEN')
+    slack_path = "https://slack.com/api/chat/postMessage"
+    query_params = {
+        'channel': 'task-notifications',
+        'text': text 
+    }
+    headers = {'Authorization': f"Bearer {SLACK_TOKEN}"}
+    requests.post(slack_path, params=query_params, headers=headers)
+
+@goals_bp.route("", methods=["GET", "POST"])
+def handle_goals():
+    if request.method == "POST": 
+        request_body = request.get_json()
+        title = request_body.get("title")
+
+        if "title" not in request_body:
+            return jsonify({"details": "Invalid data"}), 400
+
+        new_goal = Goal(title=request_body["title"])
+
+        db.session.add(new_goal)
+        db.session.commit()
+
+        committed_goal = {
+            "goal": {
+                "id": new_goal.goal_id,
+                "title": new_goal.title
+            }
+        }
+
+        return jsonify(committed_goal), 201
+
+    elif request.method == "GET":
+        goals = Goal.query.all()
+        goals_response = []
+        
+        for goal in goals:
+            goals_response.append({
+                "title": goal.title,
+                "id": goal.goal_id
+            })
+
+        return jsonify(goals_response), 200
+
+@goals_bp.route("/<goal_id>", methods=["GET", "PUT", "DELETE"])
+def handle_goal(goal_id):
+    goal = Goal.query.get_or_404(goal_id)
+    
+    if request.method =="GET":
+        if goal == None:
+            return make_response("No matching goal found"), 404
+
+        tasks = []
+
+        for item in goal.tasks:
+            tasks.append(item)
```

Comment:
You do not need to have this here. You are handling goals and tasks in the function you created `handle_goal_tasks`

```suggestion
        tasks = []

   
```